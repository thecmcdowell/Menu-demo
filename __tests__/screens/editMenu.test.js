import React from 'react'
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'
import { render, cleanup } from 'react-native-testing-library';
import EditItem from '../../app/screens/editMenuItem'

afterEach(cleanup)

jest.mock('@react-navigation/native', () => {
    const actualNav = jest.requireActual("@react-navigation/native")
    return {
        ...actualNav,
        useNavigation: () => ({
            navigate: jest.fn
        })
    }
})

jest.mock('react-native-image-picker', () => {
    return {
        launchImageLibrary: jest.fn(),
    }
})

describe('Edit item screen tests', () => {
    const mockStore = configureStore([])


    it('should display template for new item ', () => {
        const mockRoute = {
            key: '',
            name: 'Edit',
            params: null
        }

        const store = mockStore({ menu: [] })
        const rendered = render(
            <Provider store={store}>
                <EditItem route={mockRoute} />
            </Provider>
        )
        expect(rendered).toMatchSnapshot()
    })
    it('should display editable fields for exisiting items', () => {
        const mockRoute = {
            key: '',
            name: 'Edit',
            params: {
                item: {
                    id: 1,
                    image: '',
                    title: 'Taco',
                    description: 'its NOT sushi?',
                    price: '$5.99'
                }
            }
        }

        const store = mockStore({ menu: [] })
        const rendered = render(
            <Provider store={store}>
                <EditItem route={mockRoute} />
            </Provider>
        )
        expect(rendered).to
    })
})