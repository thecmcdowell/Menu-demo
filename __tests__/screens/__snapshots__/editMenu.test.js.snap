// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Edit item screen tests should display template for new item  1`] = `
<View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "opacity": 1,
      }
    }
  >
    <Image
      source={
        Object {
          "uri": "https://www.listchallenges.com/f/items/4a98113a-62e5-444a-82a1-627089b81bbb.jpg",
        }
      }
      style={
        Object {
          "height": 150,
          "width": 150,
        }
      }
    />
  </View>
  <TextInput
    onChangeText={[Function]}
    placeholder="title"
    style={
      Object {
        "borderWidth": 1,
        "height": 40,
        "margin": 12,
        "padding": 10,
      }
    }
    value=""
  />
  <TextInput
    onChangeText={[Function]}
    placeholder="description"
    style={
      Object {
        "borderWidth": 1,
        "height": 40,
        "margin": 12,
        "padding": 10,
      }
    }
    value=""
  />
  <TextInput
    keyboardType="number-pad"
    onChangeText={[Function]}
    placeholder="price"
    style={
      Object {
        "borderWidth": 1,
        "height": 40,
        "margin": 12,
        "padding": 10,
      }
    }
    value=""
  />
  <View
    accessibilityRole="button"
    accessibilityState={
      Object {
        "disabled": true,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
          Object {},
        ]
      }
    >
      <Text
        disabled={true}
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
            Object {
              "color": "#cdcdcd",
            },
          ]
        }
      >
        Create Item
      </Text>
    </View>
  </View>
</View>
`;
